classes/Dispatcher.php

```diff
@@ -315,6 +306,38 @@ private function setDefaultController($defaultController)
         $this->default_controller = $defaultController;
     }
 
+    /**
+     * Sets the front controller type.
+     */
+    private function setFrontControllerType()
+    {
+        if (defined('_PS_ADMIN_DIR_')) {
+            $this->front_controller = self::FC_ADMIN;
+        }
+        
+        if ('module' === Tools::getValue('fc')) {
+            $this->front_controller = self::FC_MODULE;
+        }
+        
+        $this->front_controller = self::FC_FRONT;
```

Comment:
`front_controller` is always `FC_FRONT`. Don't forget to return after set.

Reply:
I'll move `$this->front_controller = self::FC_FRONT;` at the beginning of the method.